src/models/volunteer.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Volunteer Model
+ */
+import { omit, pick, evolve } from 'ramda';
+import { VolunteerCollection } from './types';
+import { Users, ModelToColumn } from './user';
+import { RoleEnum } from '../auth/types';
+import { applyQueryModifiers } from './applyQueryModifiers';
+
+/*
+ * Implementation of the UserCollection type for Volunteers
+ */
+export const Volunteers: VolunteerCollection = {
+  create (a) {
+    return Users.create(a);
+  },
+
+  toColumnNames (o) {
+    return Users.toColumnNames(o);
+  },
+
+  async get (client, q = {}) {
+    const query = evolve({
+      where: Volunteers.toColumnNames,
+      whereNot: Volunteers.toColumnNames,
+    }, q);
+
+    return applyQueryModifiers(
+      client
+        .select(query.fields ? pick(query.fields, ModelToColumn) : ModelToColumn)
+        .from('user_account')
+        .leftOuterJoin('gender', 'user_account.gender_id', 'gender.gender_id')
+        .leftOuterJoin('ethnicity', 'user_account.ethnicity_id', 'ethnicity.ethnicity_id')
+        .leftOuterJoin('disability', 'user_account.disability_id', 'disability.disability_id')
+        .leftOuterJoin(
+          'user_account_access_role',
+          'user_account.user_account_id',
+          'user_account_access_role.user_account_id')
+        .whereIn(
+          'user_account_access_role.access_role_id', client('access_role')
+            .select('access_role_id')
+            .whereIn('access_role_name', [RoleEnum.VOLUNTEER, RoleEnum.VOLUNTEER_ADMIN])
+        ),
+      query
+    );
+  },
+
+  async getOne (client, query = {}) {
+    const [res] = await Volunteers.get(client, { ...query, limit: 1 });
+    return res || null;
+  },
+
+  async exists (client, query) {
+    const res = await Volunteers.getOne(client, query);
+    return res !== null;
+  },
+
+  async add (client, user) {
+    return Users.add(client, user);
+  },
+
+  async update (client, user, changes) {
+    const volunteer = Volunteers.getOne(client, { where: { id: user.id } });
```

Comment:
1. This doesn't work because `volunteer` is a `Promise<User | null>`, not a `User | null`, so it will always be truthy
2. Maybe `Volunteers.exists` would be more appropriate (I know they do mostly the same thing, but existence is actually the thing you are checking for)
3. You can inline the check into the `if`:
```ts
if (await Volunteers.exists(...)) {}
```